play 2.4 module definition

pbelmann opened the pull request:
see #43 

**Summary**:

* Play Module implementation https://www.playframework.com/documentation/2.4.x/PluginsToModules

* Removal of Play Plugin implementation as suggested in https://www.playframework.com/documentation/2.4.x/PluginsToModules

* Major version incremented due to the removal of the plugin implementation

**Usage**:

* PlayJongo must be injected:

Example:

~~~Java
public class User {

    public static PlayJongo jongo = Play.application().injector().instanceOf(PlayJongo.class);
	
    public static MongoCollection users() {
        return jongo.getCollection("Users");
    }
}
~~~

**TODO**:

If this PR is merged then the readme file must be updated.

magro commented on app/uk/co/panaxiom/playjongo/IPlayJongo.java:
```diff
@@ -0,0 +1,4 @@
+package uk.co.panaxiom.playjongo;
+
+public interface IPlayJongo {
```
What is this interface good for? Why not just bind PlayJongo.class directly?

magro commented:
Nice, LGTM! Can you update the README as part of this PR?

pbelmann commented:
@magro I removed the unnecessary interface and updated the README. 
You should update the repository description. It still states `Play 2.2 Module for Jongo http://jongo.org/`

magro commented:
@pbelmann Great, looks good! Can you please squash commits to a single one? Thanks!

For the repository description I'm missing permissions as it seems.
@alexanderjarvis Can you update the repository description (or give me the required permissions)? It still states "Play 2.2 Module for Jongo http://jongo.org/".

pbelmann commented:
@magro  I squashed my commits into two because I rebased my branch before updating the readme. I hope it's ok.
The history of this PR shows me 5 commits but there are actaully 2.
See here: 
https://github.com/pbelmann/play-jongo/commits/feature/play-2.4-module

magro commented:
Thanks! 1.0.0-jongo1.2 is on its way to maven central.

pbelmann commented:
Great! thanks!

magro commented:
@pbelmann There was an issue with the `bind(PlayJongo.class).to(PlayJongo.class)`, I changed this to `bind(PlayJongo.class).toSelf)`. This is contained in 1.0.1-jongo1.2 I just released.

pbelmann commented:
Have not noticed this bug in my play project. Anyway thanks for the fix.
